reconcile/sentry_config.py

```diff
@@ -0,0 +1,317 @@
+  state = SentryState()
+  gqlapi = gql.get_api()
+
+  # Query for users that should be in sentry
+  team_members = {}
+  result = gqlapi.query(SENTRY_USERS_QUERY)
+  github = init_github()
+  for role in result['roles']:
+    permissions = list(filter(
+      lambda p: p.get('service') == 'sentry-membership',
+      role['permissions']
+    ))
+
+    for permission in permissions:
+      # Users that should exist
+      members = []
+      team = permission["team"]
+
+      def append_github_username_members(member):
+        github_username = member.get('github_username')
+        if github_username:
+          user_info = github.get_user(login=github_username)
+          email = user_info.email
+          if email != None:
+            members.append(email)
+
+      for user in role['users']:
+        append_github_username_members(user)
+
+      for bot in role['bots']:
+        append_github_username_members(bot)
+      if team not in team_members.keys():
+        team_members[team] = members
+      else:
+        team_members[team].extend(members)
+  state.init_users_from_desired_state(team_members)
+
+  # Query for teams that should be in sentry
+  result = gqlapi.query(SENTRY_TEAMS_QUERY)
+  teams = []
+  for team in result['teams']:
+    if team in teams:
+      logging.error("sentry team %s already exists" % team)
+      continue
+    teams.append(team["name"])
+  state.init_teams(teams)
+
+  # Query for projects that should be in sentry
+  result = gqlapi.query(SENTRY_PROJECTS_QUERY)
+  projects = {}
+  for app in result['apps']:    
+    sentry_projects = app.get('sentryProjects')
+
+    if sentry_projects is None:
+      continue
+
+    for sentry_project in sentry_projects:
+      team = sentry_project["team"]["name"]
+      team_projects = []
+      for project_config in sentry_project["projects"]:
+        if project_in_project_list(project_config, projects.values()):
+          logging.error("sentry project %s already exists" % project_config["name"])
+          continue
+          
+        config = {}
+        for field in project_config.keys():
+          if project_config[field] != None:
+            config[field] = project_config[field]
+        team_projects.append(config)
+      projects[team] = team_projects
+  state.init_projects_from_desired_state(projects)
+  return state
+
+def run(dry_run=False):
+  settings = queries.get_app_interface_settings()
+  secret = {
+    "path": path,
+    "field": field,
+  }
+  host = secret_reader.read(secret, settings)
+  secret["field"] = field
+  token = secret_reader.read(secret, settings)
+
+  secret["path"] = path
+  secret["field"] = field
+  skip_users = secret_reader.read(secret, settings)
+
+  sentry_client = SentryClient(host, token)
+  current_state = fetch_current_state(sentry_client, [skip_users])
+  desired_state = fetch_desired_state()
+
+  reconciler = SentryReconciler(sentry_client, dry_run)
+  reconciler.reconcile(current_state, desired_state)
```

Comment:
missing end of line

Reply:
It doesn't appear all the files end with a blank line.  I don't care either way, just want consistency. :)